pkg/apis/v2v/v1alpha1/resourcemapping_types.go

```diff
@@ -41,6 +42,26 @@ type OvirtMappings struct {
 	DiskMappings *[]ResourceMappingItem `json:"diskMappings,omitempty"`
 }
 
+// VmwareMappings defines the mappings of vmware resources to kubevirt
+// +k8s:openapi-gen=true
+type VmwareMappings struct {
+	// NetworkMappings defines the mapping of guest network interfaces to kubevirt networks
+	// NetworkMappings.Source.Name represents the `network` field of the GuestNicInfo in vCenter
+	// NetworkMappings.Source.ID represents the `macAddress` field of the network adapter
+	// +optional
+	NetworkMappings *[]ResourceMappingItem `json:"networkMappings,omitempty"`
+
+	// StorageMappings defines the mapping of datastores to storage classes
+	// +optional
+	StorageMappings *[]ResourceMappingItem `json:"storageMappings,omitempty"`
```

Comment:
Just be careful, with sharing same class for different mappings, we had to change it[0] for oVirt, because it was not suitable for extensions, so you may hit same issue in future.

[0] https://github.com/kubevirt/vm-import-operator/pull/347/files#diff-7b282a764b7b115b17d31e0cdae066caR31

Reply:
Thanks, I'll revise.